src/lib/server/extensions/atom_feed.ml

```diff
@@ -131,19 +131,19 @@ let rec metaAttr_extract l = match l with
 let rec c_pcdata l = match l with | [] -> [] | a::r -> Tyxml_xml.pcdata a :: c_pcdata
 r
 
-let print_html5 l =
+let print_html5 e =
   let buffer = Buffer.create 500 in
-  let output = Buffer.add_string buffer in
   let encode x = fst (Xml_print.Utf8.normalize_html x) in
-  Eliom_content.Html.Printer.print_list ~encode ~output l;
+  Buffer.add_string buffer
+    (Format.asprintf "%a" (Eliom_content.Html.Printer.pp_elt ~encode ()) e);
```

Comment:
Use `Format.formatter_of_buffer` to make a formatter out with the buffer, and then `(pp_elt ~encode ()) fmt e`.